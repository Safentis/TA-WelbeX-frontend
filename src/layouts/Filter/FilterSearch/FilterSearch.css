.filter-search {
}
.filter-search__item {
  height: 100%;
}

.filter-search__input {
  height: 100%;
}

.input {
  background-color: var(--main-white-color);
  border-radius: 0.6rem;
  border: 0.3rem solid var(--main-grey-color);
  outline: none;
  padding: 1rem 1.5rem;
}

.input,
.input::placeholder {
  color: var(--main-dark-blue-color);
  font-size: 1.6rem;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}
