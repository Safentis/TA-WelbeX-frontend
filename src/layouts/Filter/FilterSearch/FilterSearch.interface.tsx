import React from 'react';

export interface Props {
  className?: string;
  value: string;
  handleSearch: (event: React.ChangeEvent) => void;
}
