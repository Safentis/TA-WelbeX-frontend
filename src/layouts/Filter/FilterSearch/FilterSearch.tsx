import React, { FC } from 'react';

import './FilterSearch.css';
import { Props } from './FilterSearch.interface';

/**
 * FilterSearch component
 * @param {string} value - required
 * @param {(event: React.ChangeEvent) => void} handleSearch - required
 * @returns {React.ReactElement} - returns react element
 */

const FilterSearch: FC<Props> = ({
  className = '',
  value,
  handleSearch,
}): React.ReactElement => {
  return (
    <ul className={'filter-search ' + className}>
      <li className="filter-search__item">
        <input
          className="input filter-search__input"
          placeholder="Search..."
          onChange={handleSearch}
          value={value}
          type="text"
        />
      </li>
    </ul>
  );
};

export default FilterSearch;
