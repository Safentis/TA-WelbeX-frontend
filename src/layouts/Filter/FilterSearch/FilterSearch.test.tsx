import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import FilterSearch from './FilterSearch';

import { Props } from './FilterSearch.interface';

const setUp = (props: Props) => {
  return shallow(<FilterSearch {...props} />);
};

describe('<FilterSearch />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
    value: '',
    handleSearch: () => {}
  }

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 ul element', () => {
      expect(component.find('ul')).toBeTruthy();
    });

    it('has 1 li element', () => {
      expect(component.find('li')).toBeTruthy();
    });

    it('has 1 input element', () => {
      expect(component.find('input')).toBeTruthy();
    });
  });
});
