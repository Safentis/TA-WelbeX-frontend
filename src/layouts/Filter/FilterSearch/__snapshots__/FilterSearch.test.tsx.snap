// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSearch /> component does match to the snapshot 1`] = `
<ul
  className="filter-search custom-style"
>
  <li
    className="filter-search__item"
  >
    <input
      className="filter-search__input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </li>
</ul>
`;
